targets/TARGET_STM/sleep.c

```diff
@@ -40,17 +40,25 @@ extern void HAL_ResumeTick(void);
 
 void hal_sleep(void)
 {
+    // Disable IRQs
+    core_util_critical_section_enter();
```

Comment:
I can confirm that since `mbed-os-5.5` I am facing this issue, which I could resolve with the patch of this PR. I never checked RTX (neither v4 nor v4) source code to have a real confirmation for my assumption, though.